src/commonMain/kotlin/io/beatmaps/api/users.kt

```diff
@@ -13,19 +13,31 @@ enum class AccountType {
 data class UserDetail(
     val id: Int,
     val name: String,
+    val description: String,
     val uniqueSet: Boolean = true,
     val hash: String? = null,
     val testplay: Boolean? = null,
     val avatar: String,
     val stats: UserStats? = null,
+    val followData: UserFollowData? = null,
     val type: AccountType,
     val email: String? = null,
     val uploadLimit: Int? = null,
+    val admin: Boolean? = null,
     val curator: Boolean? = null,
     val verifiedMapper: Boolean = false,
-    val following: Boolean? = null
 ) { companion object }
 @Serializable
+data class FollowerData(
```

Comment:
Worth having a common type these both inherit from as this is all stuff in UserDetail except `maps`

Reply:
There's no practical use for that right now. I think doing this would only require more code, since the constructors of the derived classes would still need to have every property listed.

Reply:
I actually only mentioned this because I saw a practical use (the duplicated code in userTitles)

Reply:
Wouldn't that still need separate functions because of `maps` vs `stats.totalMaps`?

Reply:
Not necessarily, if you had a function to pull that property out it wouldn't change how they serialise but that method would work.

Reply:
You are right though data class inheritance is a bit painful.